src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/SegmentValidations/ScriptSegmentValidationData.cs

```diff
@@ -0,0 +1,200 @@
+                                if (type.FullName != null && type.FullName.Equals(classFullName))
+                                {
+                                    t = type;
+                                    CachedTypes[classFullName] = t;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+
+                    // Make sure this type is not null and is inheriting from RGValidateBehaviour
+                    if (t == null)
+                    {
+                        _error = $"Regression Games could not load Bot Segment Validation Script for Type - {classFullName}. Type was not found in any assembly in the current runtime.";
+                        RGDebug.LogError(_error);
+                    }
+                    else if (!typeof(RGValidateBehaviour).IsAssignableFrom(t))
+                    {
+                        _error = $"Regression Games could not load Bot Segment Validation Script for Type - {classFullName}. This Type does not inherit from RGValidateBehaviour.";
+                        RGDebug.LogError(_error);
+                    }
+                    else
+                    {
+                        _typeToCreate = t;
+                    }
+
+                    _readyToCreate = true;
+                }).Start();
+            }
+        }
+
+        public void ProcessValidation(int segmentNumber)
+        {
+
+            if (!_isStopped)
+            {
+                if (_readyToCreate)
+                {
+                    _readyToCreate = false;
+                    if (_typeToCreate != null)
+                    {
+                        // load behaviour and set as a child of the playback controller
+                        var pbController = Object.FindObjectOfType<BotSegmentsPlaybackController>();
+                        _myGameObject = new GameObject($"RGValidate_{classFullName}")
+                        {
+                            transform =
+                            {
+                                parent = pbController.transform
+                            }
+                        };
+
+                        // Attach our validation script
+                        _myValidationBehaviour = _myGameObject.AddComponent(_typeToCreate) as RGValidateBehaviour;
+                    }
+                    else
+                    {
+                        RGDebug.LogError("Could not load type for validation script");
+                        _isStopped = true;
+                    }
+                }
+
+                if (_myGameObject != null)
+                {
+                    if (!_myGameObject.TryGetComponent(_typeToCreate, out _))
+                    {
+                        ((IRGSegmentValidationData)this).StopValidation(segmentNumber);
+                    }
+
+                    // TODO(vontell): How do I want to support rogue scripts that keep running?
```

Comment:
Ideas here? I think I am going to add a timeout option to these validation scripts...

Reply:
I added a timeout option.